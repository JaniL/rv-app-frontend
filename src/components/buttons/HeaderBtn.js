import React from 'react';
import classNames from 'classnames';
import './styles/Button.css';
import './styles/HeaderBtn.css';
import Loader from './../loaders/Loader';
import PropTypes from 'prop-types';

const HeaderBtn = ({ onClick, children, fill, hover, loader, ...props }) => {
    const className = classNames(
        'btn',
        fill ? 'headerbtn-fill' : 'headerbtn',
        {
            'headerbtn-fill-hover': hover && fill,
            'headerbtn-hover': hover && !fill
        }
    );
    return (
        <button {...props} onClick={onClick} className={className}>
            {!loader ? (
                <span className="btnContent">{children}</span>
            ) : (
                <Loader />
            )}
        </button>
    );
};

HeaderBtn.propTypes = {
    /** Click handler. */
    onClick: PropTypes.func,
    /** Button fill. If set to `true`, the button's colors will "invert". */
    fill: PropTypes.bool,
    /** Hover effect. If set to `true`, the button will have a hover effect. */
    hover: PropTypes.bool,
    /** Loader effect. If set to `true`, will show a CSS loader instead of button text. */
    loader: PropTypes.bool
};

HeaderBtn.defaultProps = {
    onClick: null,
    fill: false,
    hover: false,
    loader: false
};

export default HeaderBtn;
