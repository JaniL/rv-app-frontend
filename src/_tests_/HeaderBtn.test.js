import React from 'react';
import ReactDOM from 'react-dom';
import HeaderBtn from '../components/buttons/HeaderBtn';
import Enzyme from 'enzyme';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<HeaderBtn />, div);
    const cmpnt = renderer.create(<HeaderBtn />).toJSON();
    expect(cmpnt).toMatchSnapshot();
});
