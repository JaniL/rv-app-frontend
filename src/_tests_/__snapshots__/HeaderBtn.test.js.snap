// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<button
  className="btn headerbtn"
  onClick={null}
>
  <span
    className="btnContent"
  />
</button>
`;
